File: 07-langs_particularities/2-java/src/main/java/tv/codely/contexts/rrss/posts/domain/PostDoesNotExistError.java
package tv.codely.contexts.rrss.posts.domain;

import tv.codely.contexts.shared.domain.DomainError;

public class PostDoesNotExistError extends DomainError {
    public PostDoesNotExistError(String id) {
        super(String.format("The post <%s> does not exist", id));
    }
}

File: 08-ts_types/1-fp_ts/src/contexts/rrss/posts/domain/PostDoesNotExistError.ts
import { DomainError } from "../../../shared/domain/DomainError";

export class PostDoesNotExistError extends DomainError {
	readonly type = "PostDoesNotExistError";
	readonly message = `The post ${this.id} does not exist`;

	constructor(public readonly id: string) {
		super();
	}
}
